images/cassandra/tests/main.tf

```diff
@@ -0,0 +1,41 @@
+terraform {
+  required_providers {
+    oci = { source = "chainguard-dev/oci" }
+    kubernetes = {
+      source  = "hashicorp/kubernetes"
+      version = "~> 2.11.0"
+    }
+  }
+}
+
+provider "kubernetes" {
+  config_path    = "~/.kube/config"
+  config_context = "kind-kind"
+}
+
+variable "digest" {
+  description = "The image digest to run tests over."
+}
+
+data "oci_string" "ref" { input = var.digest }
+
+data "oci_exec_test" "test" {
+  for_each = toset(fileset(path.module, "*.sh"))
+
+  digest = var.digest
+  script = "${path.module}/${each.value}"
+}
+
+resource "kubernetes_manifest" "prerequisitesS" {
+  for_each = fileset(path.module, "manifests/*.yaml")
+  manifest = yamldecode(file("${path.module}/${each.value}"))
+}
+
+resource "kubernetes_manifest" "stateful_set" {
```

Comment:
This line is failing:

```
  │ Error: Failed to determine GroupVersionResource for manifest
  ╵
  │ 
  │   with module.cassandra.module.test-latest.kubernetes_manifest.stateful_set,
  │   on images/cassandra/tests/main.tf line 34, in resource "kubernetes_manifest" "stateful_set":
  │   34: resource "kubernetes_manifest" "stateful_set" {
  │ 
  │ unmarshaling unknown values is not supported
```

Reply:
Guessing you need to declare what this `depends_on`... something?

Reply:
I don't know enough about terraform to say exactly the problem but I would try to copy the pattern calico is using to see if that helps: https://github.com/chainguard-images/images/blob/e0320db17ac10f5303c5985228572af0f58dd256/images/calico/kind-tests/main.tf#L34-L58

Reply:
Yes, thanks a lot @jonjohnsonjr, your comment and [this issue](https://github.com/hashicorp/terraform-provider-kubernetes/issues/1380#issuecomment-967022975) suggestion helped a lot, I guess it should be fixed now 🤞